README.md

```diff
@@ -1 +1,102 @@
+<div>
+    <img alt="state-mate" src="https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fraw.githubusercontent.com%2Flidofinance%2Fstate-mate%2Fmain%2Fpackage.json&query=%24.version&label=state-mate&labelColor=white&color=green"/>
+    <img alt="Node.js" src="https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fraw.githubusercontent.com%2Flidofinance%2Fstate-mate%2Fmain%2Fpackage.json&query=%24.engines.node&style=flat&label=node.js&labelColor=rgb(62%2C%20109%2C%2026)&color=white"/>
+    <img alt="GitHub license" src="https://img.shields.io/github/license/lidofinance/state-mate?labelColor=orange&color=white"/>
+</div>
+
 # state-mate
+
+state-mate is a simple tool that validates contracts' states against a concise YAML-based description. Run state-mate to verify deploy outcome, current state, access control and more.
+
+state-mate accepts a yaml file that includes contract addresses, view functions and their expected results. It calls each function and compares the output to the expected result.
+
+## ✨ Features
+
+- state (non-mutable functions result) verification,
+- automatically validates that all functions covered,
+- supports any EVM network,
+- easily configurable checks,
+- CI-friendly,
+- uses yaml config (env variables, variables, comments).
+
+## ⚡ Getting started
+
+### Requirements
+
+- git
+- Node.js >=20,
+- yarn 1.x
+
+### Usage
+
+1. Install dependencies
+
+```sh
+yarn install
+```
+
+2. Specify RPC endpoints for your target networks, e.g.
+
+```sh
+export L1_MAINNET_RPC_URL=%YOUR_RPC_URL%
+```
+
+3. Start the program
+
+```sh
+yarn start path/to/config.yaml
+```
+
+### Configuration
+
+Config is a yaml file that contains all the required addresses, parameters, view functions with their expected results for verification. The outline of the config is given below,
+
+```yaml
+# Sample config
+
+parameters:
+  # List of parameters
+  - &MY_PARAMETER 42
+
+misc:
+  # Misc variables
+  - &FOO "foo"
+
+deployed:
+  # Contract addresses
+  - &myContract "0x0000000000000000000000000000000000000001"
+  - &adminMultisig "0x0000000000000000000000000000000000000002"
+
+roles:
+  # ACL checks
+  - &DEFAULT_ADMIN_ROLE "0x0000000000000000000000000000000000000000000000000000000000000000"
+
+l1:
+  rpcUrl: L1_MAINNET_RPC_URL # env variable
+  contracts:
+    myContract:
+      name: "myContract"
+      address: *myContract
+      implementation: "%implementation address%"
+      proxyChecks:
+        proxy__getAdmin: *adminMultisig
+      checks:
+        # list of view functions and expected results
+        getMyParameter: *MY_PARAMETER
+        getFoo: *FOO
+      ozAcl:
+        *DEFAULT_ADMIN_ROLE : [*adminMultisig]
+```
+
+### ABIs
+
+All requried ABIs must be located in the same directory as the config and placed under `abi` folder. See [configs](/configs/).
```

Comment:
Shall we retrieve abis from Etherscan (even though it would provoke a bit of complication for proxies)?

Reply:
For now, there is no such feature, so the doc reflects it. We'll update this section once it's delivered

Reply:
Agree, filed an issue #8